internal/abci/abci.go

```diff
@@ -164,6 +172,29 @@ func NewAbciApp(ctx context.Context, cfg *AbciConfig, snapshotter SnapshotModule
 	if height == -1 {
 		height = 0 // negative means first start (no state table yet), but need non-negative for below logic
 	}
+
+	if bytes.Equal(appHash, []byte{0x42}) {
+		// This is an interesting corner case where the node crashes after the commit of block execution
+		// state but before the commit of the app hash. The chain state will have the app hash as "42".
+		// CometBFT seems to persist the FinalizeBlock responses, so during restart it calls "Commit"
+		// directly without calling FinalizeBlock and assumes that the application has the correct state
+		// persisted, whereas the application has the app hash as "42". This will cause the current block
+		// to be committed with an invalid app hash (sentry node blocksyncing will fail at this point).
```

Comment:
The sentry node comment is confusing.  Any node block sync will fail, including this one if it tries to resync.  You are alluding to the case where if this nodes is the only validator, there will be no other nodes with which it will disagree to cause a consensus failure, resulting in silent corruption that will only be detected on resync (or other node sync)